### PR TITLE
broken ledge grabs

### PR DESCRIPTION
**Pull request by ScottDikowitz**

### CODE REVIEW

File: src/game/Mario.js [hunk 1]
```diff
@@ -775,6 +796,7 @@ const update_mario_geometry_inputs = (m) => {
     m.waterLevel = -20000.0 //find_water_level(m->pos[0], m->pos[2]);
 
     if (m.floor) {
+        if (!m.floor.normal) m.floor.normal = {z: 0, x: 0};
```
Comment (ScottDikowitz) [inline src/game/Mario.js:799]: broken pls fix

Comment (sm64js) [reply src/game/Mario.js:799]: So, I added throw error here if m.floor.normal is null,  and so far that error has never gone off.   Any ideas?  Do you know what conditions caused problems here?

File: src/game/Camera.js [hunk 2]
```diff
@@ -203,7 +203,7 @@ class Camera {
         ObjectListProc.gCheckingSurfaceCollisionsForCamera = true
 
         if (SurfaceCollision.find_floor(this.gPlayerCameraState.pos[0], this.gPlayerCameraState.pos[1] + 10.0, this.gPlayerCameraState.pos[2], surf) != -11000) {
-            pg.currFloorType = surf.floor.type
+            pg.currFloorType = surf.floor?.type ?? 0
```
[the remark this answers is not in the document]
Comment (sm64js) [reply src/game/Camera.js:206]: Same thing here

So, I added throw error here if surf.floor.type is NaN,  and so far that error has never gone off.   Any ideas?  Do you know what conditions caused problems here?

File: src/game/Mario.js (hunk 1 above)
Comment (ScottDikowitz) [reply src/game/Mario.js:799]: not sure, might have been fixed with your sin/cos changes!

**Pull request merged**